hardware/mifare/mifare_classic.ksy

```diff
@@ -0,0 +1,243 @@
+meta:
+  id: mifare_classic
+  endian: le
+  file-extension: mfd
+  title: "Mifare Classic RFID tag dump"
+  license: BSD-2-Clause
+doc-ref: |
+  https://github.com/nfc-tools/libnfc
+  https://www.nxp.com/docs/en/data-sheet/MF1S70YYX_V1.pdf
+seq:
+  - id: sectors
+    type: sector(_index)
+    repeat: eos
+types:
+  key:
+    seq:
+      - id: key
+        size: 6
+  sector:
+    params:
+      - id: idx
+        type: u1
+    seq:
+      - id: manufacturer
+        type: manufacturer
+        if: has_manufacturer
+      
+      - id: data_
```

Comment:
Having `data_` and `data` will break for all languages with non-underscore case, i.e. C#, Java, JavaScript, etc.

Reply:
I guess we should keep trailing underscores. It's a pretty little addition for name transforming functions.

Reply:
No, we should not. This would raise tons of complaints from style checkers in all these languages.

Reply:
Gmmm, maybe to replace the final underscore with `Internal<number here for uniquiness>' ?

Reply:
Sure, that will work.

Reply:
fixed. But still, we really should make the ksy names with trailing underscore workable cross-language.